hail/python/hailtop/config/deploy_config.py

```diff
@@ -6,6 +6,20 @@
 log = logging.getLogger('gear')
 
 
+def get_conf_dir():
+    xdg_conf_home = os.environ.get('XDG_CONFIG_HOME', os.path.expanduser('~/.config'))
+    hail_conf = os.path.join(xdg_conf_home, 'hail')
+    if os.path.isdir(hail_conf):
+        return hail_conf
+    dot_hail = os.path.expanduser('~/.hail')
+    if not os.path.exists(dot_hail):
```

Comment:
Could we do the inverse?

Check whether the current paths are filled, and then fallback to XDG_CONFIG_HOME if not? That seems like it would prevent
> If ~/.hail already exists on your system, this should not break you as
long as $XDG_CONFIG_HOME/hail or ~/.config/hail also do not exist.

Reply:
I'll check if the tokens file exists. Just checking for the old path first defeats the purpose of the change. I want to change the order of the search.

Reply:
I don't see the problem. I thought your goal was to avoid having to clutter up your home directory. If so, keeping the order and appending an $XDG_CONFIG_HOME check fixes the issue, while also preserving as much of the existing behavior as possible for users that have come to rely on it. 

Also, why are we adding ~/.config?

Reply:
This rule still seems too complicated:

```python
def get_conf_dir():
     conf_folder = os.path.expanduser('~/.hail')
     if os.path.exists(conf_folder):
         return conf_folder

     xdg_conf_home = os.environ.get('XDG_CONFIG_HOME', os.path.expanduser('~/.config'))
     conf_folder = os.path.join(xdg_conf_home, 'hail')
     if os.path.exists(conf_folder):
         return conf_folder

    raise FileNotFoundError(
    errno.ENOENT, os.strerror(errno.ENOENT), "Hail config folder expect at ~/.hail, ~/.config or $XDG_CONFIG_HOME/hail. No matching directories found" )
```

I don't want to rely on the name of the token file. We've cycled through several names for tokens; a future change will have non-obvious behavior for users that have `~/.hail/tokens.json`.